subprojects/java21/src/test/java/org/mockito/internal/stubbing/answers/DeepStubReturnsEnumJava21Test.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2023 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.internal.stubbing.answers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.*;
+
+import org.junit.Test;
+import org.mockito.exceptions.base.MockitoException;
+
+public class DeepStubReturnsEnumJava21Test {
+
+    @Test
+    public void cant_mock_enum_class_in_Java21_Issue_2984() {
+        assertThatThrownBy(
+                        () -> {
+                            mock(TestEnum.class);
+                        })
+                .isInstanceOf(MockitoException.class)
+                .hasMessage(
+                        "\nMockito cannot mock this class: class org.mockito.internal.stubbing.answers.DeepStubReturnsEnumJava21Test$TestEnum.\n"
+                                + "Sealed abstract enums can't be mocked. Since Java 15 abstract enums are declared sealed, which prevents mocking.\n"
+                                + "You can still return an existing enum literal from a stubbed method call.")
+                .hasCauseInstanceOf(MockitoException.class);
+    }
+
+    @Test
+    public void cant_mock_enum_class_as_deep_stub_in_Java21_Issue_2984() {
+        assertThatThrownBy(
+                        () -> {
+                            mock(TestEnum.class, RETURNS_DEEP_STUBS);
+                        })
+                .isInstanceOf(MockitoException.class)
+                .hasMessage(
+                        "\nMockito cannot mock this class: class org.mockito.internal.stubbing.answers.DeepStubReturnsEnumJava21Test$TestEnum.\n"
+                                + "Sealed abstract enums can't be mocked. Since Java 15 abstract enums are declared sealed, which prevents mocking.\n"
+                                + "You can still return an existing enum literal from a stubbed method call.")
+                .hasCauseInstanceOf(MockitoException.class);
+    }
+
+    @Test
+    public void deep_stub_cant_mock_enum_with_abstract_method_in_Java21_Issue_2984() {
+        final var mock = mock(TestClass.class, RETURNS_DEEP_STUBS);
+        assertThatThrownBy(
+                        () -> {
+                            mock.getTestEnum();
+                        })
+                .isInstanceOf(MockitoException.class)
+                .hasMessageContaining("Sealed abstract enums can't be mocked.");
```

Comment:
Ah here you did it already. Yeah basically this, but with the other two test methods